### DISCUSSION TITLE
How are the GA matrices / DiracGamma defined?

### DISCUSSION THREAD

**Discussion by rmastand** [FeynCalc/feyncalc#117] [Q&A]
Hello,

I am trying to create a custom vertex to use in FeynCalc that is based on the individual gamma matrices (so I need access to Gamma1, Gamma2, etc as opposed to just the generic GA[mu]). I think the easiest thing is to write a function for this vertex from scratch, but to do this it would be very helpful to know the explicit definition of DiracGamma in Mathematica so that my new vertex can work smoothly with the existing FeynCalc functions. I've searched the source code but I just cannot seem to find where DiracGamma is defined -- where should I be looking?

Thank you!

**Comment by vsht** [comment]
Hi,

I'm afraid that your vertex will not work smoothly with the existing FeynCalc functions OOTB,
since Dirac algebra simplification for explicit special indices is not implemented. So essentially
you would have to implement a custom routine that does that for you.


I'm also not sure what you mean with "defined". [This](https://github.com/FeynCalc/feyncalc/blob/069907f5d2a069612e96f11c6fab12f008bae9eb/FeynCalc/Shared/SharedObjects.m#L1277)?

BTW, Gamma1 and Gamma2 have nothing to do with the Dirac matrices. You can look this
stuff up in the [online help](https://feyncalc.github.io/FeynCalcBook/guide/FeynCalc.html)

Integers inserted into Lorentz index slots are converted to ExplicitLorentzIndex, e.g.

{GA[1], GA[2], GA[3], GA[0]}
% // FCI // StandardForm

So having a vertex that depends on say GA[2] is not an issue, but simplifying such expressions
requires extra work.

Cheers,
Vladyslav

**Reply by vsht** [reply]
On the other hand, I had this explicit index algebra feature on my ToDo list for quite some time, so perhaps
it is a good opportunity to test it.  The commit a5395a7 add such manipulations to the dev version, although
not they are not so well tested. But in principle things like
```
DiracSimplify[GA[1, 3, 2, mu, 0, 3, 1, mu, 2, 1, 2, 3]]
```
should now work.
